fs/ggml/ggml.go

```diff
@@ -214,6 +214,9 @@ type Tensor struct {
 	// Shape is the number of elements in each dimension
 	Shape []uint64 `json:"shape"`
 
+	// if BufCh is populated, this will be used for writing
+	// else WriterTo is used
+	BufCh       chan []byte `json:"-"`
```

Comment:
Similar behaviour should be achievable with an [`io.Pipe`](https://pkg.go.dev/io#PipeWriter). While it doesn't implement `io.WriterTo`, it does implement `io.Writer`

Reply:
+ having a channel as part of the Tensor struct feels like the wrong abstraction

Reply:
I'll refactor this to replace io.WriterTo with io.Pipe.